tslearn/nonmyopic.py

```diff
@@ -0,0 +1,509 @@
+                                self.clusters_["ck_cm{0}".format(current_cluster)][-1][
+                                    0
+                                ][1]
+                                / column_sum[0],
+                            ],
+                            [
+                                self.clusters_["ck_cm{0}".format(current_cluster)][-1][
+                                    1
+                                ][0]
+                                / column_sum[1],
+                                self.clusters_["ck_cm{0}".format(current_cluster)][-1][
+                                    1
+                                ][1]
+                                / column_sum[1],
+                            ],
+                        ]
+                    )
+                    self.pyhatyck_["pyhatycks{0}".format(current_cluster)].append(
+                        matrix_proba_current_cluster_class
+                    )
+
+    def get_avg_dist(self, X):
+        """
+        Compute the average euclidean distance of a given time series to the centroid of each cluster of the model
+
+        Parameters
+        ----------
+        X: Vector
+            A time series that may  be truncated
+
+        Returns
+        -------
+        Float : the average euclidean distance of the time series to the centroid of each cluster of the model
+        Float : the minimum distance between the centroid of a cluster and the time series
+        """
+
+        centroids = self.cluster_.cluster_centers_[:, :len(X)]
+        cluster_distances = np.array(list(map(lambda current_centre: np.linalg.norm(X - current_centre), centroids)))
+        sum_distances = cluster_distances.sum()
+        minimum_distance = cluster_distances.min()
+        return sum_distances / self.number_cluster, minimum_distance
+
+    def get_dist(self, X, wanted_cluster):
+        """
+        Compute the euclidean distance between a time series xt and the centroid of a cluster of the model
+        Parameters
+        ----------
+        X: vector
+            a time series that may be truncated
+        wanted_cluster: int
+            the number of the cluster for which the distance is wanted
+
+        Returns
+        -------
+        Float: the expected euclidean distance
+        """
+
+        euclidean_dist = np.linalg.norm(
+            X - self.cluster_.cluster_centers_[wanted_cluster, : len(X)]
+        )
+
+        return euclidean_dist
+
+    def get_cluster_prob(self, X):
+        """
+        This function computes the probabilities of the time series xt to be in a cluster of the model
+
+        Parameters
+        ----------
+        X: vector
+            a time series that may be truncated
+
+        Returns
+        -------
+        vector : the probabilities of the given time series to be in the clusters of the model
+        """
+        average_distance, minimum_distance = self.get_avg_dist(X)
```

Comment:
In this call you compute distances to all cluster centers, and you will recompute it later in the loop, that's a pity. I'd rather first get all distances, and from that list extract all information (`avg_dist`, `min_dist`, and all individual distances).